Source/Fuse.Reactive.Expressions/IsDefined.uno

```diff
@@ -117,4 +117,88 @@ namespace Fuse.Reactive
 		
 		protected override bool IsOperandOptional { get { return true; } }
 	}
+	
+	[UXFunction("nonNull")]
+	/** Returns the value if it isn't null otherwise doesn't evaluate.
+	
+		This is a special use function in cases where you need to deal temporarily with null values that later become non-null. Instead of the null creating errors in an expression chain, this causes the expression to not evaluate at all.
+		
+		@advanced
+	*/
+	public sealed class NonNull : Expression
+	{
+		public Expression Operand { get; private set; }
+		
+		[UXConstructor]
+		public NonNull([UXParameter("Operand")] Expression operand)
+		{
+			Operand = operand;
+		}
+
+		public override IDisposable Subscribe(IContext context, IListener listener)
+		{
+			return Subscription.Create(this, context, listener);
+		}
+
+		class Subscription: InnerListener
+		{
+			readonly NonNull _hv;
+			IDisposable _opSub;
+			IListener _listener;
+			bool _hasValue;
+
+			protected Subscription(NonNull hv, IListener listener)
+			{
+				_hv = hv;
+				_listener = listener;
```

Comment:
Should we validate that `hv` and `listener` is non-null here?

Reply:
We could, but since these are private classes with private construction, whereby the outer `Subscribe` method is typically called by the UX compiler code, the chance of a null is nearly 0 here.  IF anything the outer function should check, and then I'd prefer to change all of the Subscribe methods at once (there's a lot).  So I don't think it's part of this PR at least.

Reply:
Sure, that makes sense. I just generally like doing null-checks in constructors as it clearly define the contract of the object, and is virtually free (compared to object allocations etc)